src/extra/AutoUi/Collection/Filters.tsx

```diff
@@ -24,6 +26,47 @@ export const Filters = <T extends AutoUIBaseResource<T>>({
 	renderMode,
 }: FiltersProps<T>) => {
 	const history = useHistory();
+	const { analyticsClient } = useAnalytics();
+
+	React.useEffect(() => {
+		if (!analyticsClient || !filters?.length) {
+			return;
+		}
+		const analyticsObj: {
+			searchText: string;
+			field: string[];
+			operator: string[];
+			value: string[];
+		} = {
+			searchText: '',
+			field: [],
+			operator: [],
+			value: [],
+		};
+
+		filters.forEach((filter) => {
+			if (filter.title === FULL_TEXT_SLUG) {
+				// @ts-expect-error
+				analyticsObj.searchText = filter.anyOf[0].description.replace(
+					'Any field contains ',
+					'',
+				);
+			} else {
+				filter.anyOf?.forEach((item: JSONSchema) => {
+					const filterObj: { name: string; operator: string; value: string } =
+						item.description && JSON.parse(item.description);
+					if (filterObj) {
+						analyticsObj.field.push(filterObj.name);
+						analyticsObj.operator.push(filterObj.operator);
+						analyticsObj.value.push(filterObj.value);
+					}
+				});
+			}
+		});
+
+		analyticsClient.track('Filters Action', analyticsObj);
```

Comment:
Ah I realized that only the web tracker has this `prefix` defined in the initialization. See [the module's code](https://github.com/balena-io-modules/analytics-client/blob/23064da75027d9013cc34fc8364f2577cfe564f5/src/web.ts#L12) and how [Hub does it now](https://github.com/balena-io/balenahub/blob/cad538072b330492e949e6c287dc30460ea6e577/src/api/analytics.ts#L29). When we use the regular method `client.track` (not web tracker one), we prepend the prefix ourselves. So this becomes:
```js
analyticsClient.track('[Hub] Filters Action', analyticsObj);
```

If we'd like to keep prepending the component name, we need to make it a configuration option. So the regular client also prepends such prefix to each event.

Reply:
I like the prefix method and I'd like to keep it. What do you think @laziob?